.editorconfig

```diff
@@ -0,0 +1,28 @@
+# EditorConfig: https://editorconfig.org/
+root = true
+
+[*.{c,h}]
+charset = utf-8
+end_of_line = crlf
+indent_style = space
+insert_final_newline = true
+trim_trailing_whitespace = true
+
+# Makefiles
+[{{M,m}akefile,descrip.mms}]
+trim_trailing_whitespace = true
+indent_style = tab
+end_of_line = crlf
+insert_final_newline = true
+
+# Other text files
+[*.{txt,ini,md,simh,.editorconfig}]
+charset = utf-8
+end_of_line = crlf
+trim_trailing_whitespace = true
+
+# Git commit messages
+[{COMMIT_EDITMSG,MERGE_MSG}]
+trim_trailing_whitespace = true
+max_line_length = 80

```

Comment:
*Nota bene:* The prevailing line-limit for Git commit messages is 72 characters,[^1] which leaves room for at least one level of indentation (expressed as an 8-column tab). This gives breathing room for subcommands that display indented logs.

[^1]: https://cbea.ms/git-commit/#seven-rules § <i>"The seven rules of a great Git commit message"</i>

Reply:
The line length limit of 80 is what I used when merging commits for about 10 years.

I was fuzzy about having the commit message section completely.

What git clients leverage information in an .editorconfig?

Reply:
> What Git clients leverage information in an `.editorconfig?`

I'm not sure what you mean by "Git client", but literally any text-editor [git-commit(1)](https://git-scm.com/docs/git-commit#_environment_and_configuration_variables) is configured to use will be editing `.git/COMMIT_EDITMSG`, meaning that its settings are potentially influenced by the existence of an `.editorconfig` file.

> … 80 is what I used when merging commits for about 10 years.

Well, you've been doing it wrong for the past decade. But hey, you did better than I did: I wasn't even using version control until mid-2013, following a near-decade of sloppily "backing up" projects to ZIP files kept on a single 512 MBs thumb drive (which is now bricked). Oh, and my commit messages looked like [this] before I discovered that blog-post I linked you to.

So... 80 columns instead of 72 columns for 10 years actually ain't as bad as it sounds. 😉

[this]: [Alhadis/Snippets@a859db1](https://github.com/Alhadis/Snippets/commit/a859db1a5cec11c50cca8d0946c1f89fc8daa5eb)